src/settings.rs

```diff
@@ -1,160 +1,336 @@
+    }
+}
+
+#[repr(usize)]
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone, Copy)]
+#[serde(rename_all = "lowercase")]
+pub enum MidiLoading {
+    Ram = 0,
+    Live = 1,
+}
+
+impl MidiLoading {
+    #[inline]
+    pub const fn as_str(self) -> &'static str {
+        match self {
+            MidiLoading::Ram => "In RAM",
+            MidiLoading::Live => "Live",
+        }
+    }
+}
+
+impl FromStr for MidiLoading {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "ram" | "RAM" | "Ram" => Ok(MidiLoading::Ram),
+            "live" | "Live" => Ok(MidiLoading::Live),
+            s => Err(format!(
+                "{} was not expected. Expected one of `ram`, or `live`",
+                s
+            )),
+        }
+    }
+}
+
+#[repr(usize)]
+#[derive(Debug, Serialize, Deserialize, Clone, Copy, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum Synth {
+    XSynth = 0,
+    Kdmapi = 1,
+}
+
+impl Synth {
+    #[inline]
+    pub const fn as_str(self) -> &'static str {
+        match self {
+            Synth::XSynth => "XSynth",
+            Synth::Kdmapi => "KDMAPI",
+        }
+    }
+}
+
+impl FromStr for Synth {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "xsynth" | "XSynth" => Ok(Synth::XSynth),
+            "kdmapi" | "KDMPAI" => Ok(Synth::Kdmapi),
+            s => Err(format!(
+                "{} was not expected. Expected one of `xsynth`, or `kdmapi`",
+                s
+            )),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct VisualSettings {
+    #[serde(with = "color32_serde")]
     pub bg_color: Color32,
+    #[serde(with = "color32_serde")]
     pub bar_color: Color32,
+    pub show_top_pannel: bool,
+    pub show_statistics: bool,
+    pub fullscreen: bool,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct MidiSettings {
+    pub note_speed: f64,
     pub random_colors: bool,
-    pub sfz_path: String,
+    #[serde(with = "range_serde")]
     pub key_range: RangeInclusive<u8>,
-    pub midi_loading: usize,
+    pub midi_loading: MidiLoading,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct SynthSettings {
+    pub synth: Synth,
     pub buffer_ms: f64,
+    pub sfz_path: String,
     pub limit_layers: bool,
     pub layer_count: usize,
+    #[serde(with = "range_serde")]
+    pub vel_ignore: RangeInclusive<u8>,
     pub fade_out_kill: bool,
     pub linear_envelope: bool,
     pub use_effects: bool,
-    pub vel_ignore: RangeInclusive<u8>,
-    pub synth: usize,
+}
+
+#[derive(Debug, Deserialize, Serialize)]
+pub struct WasabiSettings {
+    pub synth: SynthSettings,
+    pub midi: MidiSettings,
+    pub visual: VisualSettings,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub load_midi_file: Option<String>,
 }
 
 impl Default for WasabiSettings {
     fn default() -> Self {
```

Comment:
I think it would be better to make the `Default` implementation for each struct separately.